src/main/groovy/org/labkey/gradle/plugin/Jsp.groovy

```diff
@@ -173,6 +173,10 @@ class Jsp implements Plugin<Project>
                task.group = GroupNames.JSP
                task.description = "compile jsp files into Java classes"
 
+               if (project.hasProperty('apacheTomcatVersion'))
+                   task.inputs.property 'apacheTomcatVersion', project.property('apacheTomcatVersion')
+               else
+                   task.outputs.cacheIf { false } // Just don't cache if we can't be sure of the Jasper version
```

Comment:
I'm not sure when this else case would happen, or if we'd actually be able to do much without a tomcat version, but not caching seems reasonable in that case.